### PR TITLE
Update index.md

### PR DESCRIPTION
**Pull request by minhthin1028**
fix links/text

### CODE REVIEW

File: docs/src/index.md [hunk 1]
```diff
@@ -16,7 +16,7 @@ Welcome to the documentation page for Bloqade, a &nbsp;
...
 
-Currently, Bloqade enables the easy design and fast execution of quantum dynamics in the analog mode,  based on the neutral-atom quantum computing architecture. Besides fast full Hilbert-space simulation on CPUs, the main features include the design of arbitrary-layout quantum registers ([Lattices](@ref)), easy waveform generation ([Waveforms](@ref)), the simulation in subspace constrained by the Rydberg blockade ([subspace](@ref)), faster GPU-accelerated simulation ([CUDA Acceleration](@ref)), and more.
+Currently, Bloqade enables the easy design and fast execution of quantum dynamics in the analog mode,  based on the neutral-atom quantum computing architecture. Besides fast full Hilbert-space simulation on CPUs, the main features include the design of arbitrary-layout quantum registers ([Lattices](@ref)), easy waveform generation ([Waveforms](@ref)), the simulation in subspace constrained by the Rydberg blockade ([Working with Subspace](@ref)), faster GPU-accelerated simulation ([CUDA Acceleration](https://queracomputing.github.io/Bloqade.jl/dev/cuda/)), and more.
```
Comment (Roger-luo) [inline docs/src/index.md:19]:
```suggestion
Currently, Bloqade enables the easy design and fast execution of quantum dynamics in the analog mode,  based on the neutral-atom quantum computing architecture. Besides fast full Hilbert-space simulation on CPUs, the main features include the design of arbitrary-layout quantum registers ([Lattices](@ref)), easy waveform generation ([Waveforms](@ref)), the simulation in subspace constrained by the Rydberg blockade ([Working with Subspace](@ref subspace)), faster GPU-accelerated simulation ([CUDA Acceleration](@ref cuda)), and more.
```

also need to add `@id cuda` in CUDA acceleration section

**Review by Roger-luo — CHANGES_REQUESTED** [review]
let's remove all the global links and replace them with markup references

File: docs/src/emulation.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ end
 ```
 
 You can use any function on the `reg` object.  For calculating observables, 
-please see the [Observables](@ref) section.
+please see the [Observables](https://queracomputing.github.io/Bloqade.jl/dev/registers/) section.
```
[the remark this answers is not in the document]
Comment (minhthin1028) [reply docs/src/emulation.md]: Yes, all the links should be fixed as references/ids now.

File: docs/src/emulation.md [hunk 3]
```diff
@@ -67,7 +67,7 @@ end
 ```
 
 You can use any function on the `reg` object.  For calculating observables, 
-please see the [Observables](@ref) section.
+please see the [Registers and Observables](@ref) section.
```
Comment (Roger-luo) [inline docs/src/emulation.md:70]:
```suggestion
please see the [Registers and Observables](@ref observables) section.
```

File: docs/src/registers.md [hunk 4]
```diff
@@ -64,7 +64,7 @@ apply!(reg, put(1=>X))
 ```
 
 For a more detailed introduction of the register interface, please
-refer to [Yao:Array Registers](https://docs.yaoquantum.org/dev/man/array_registers.html) and [Yao:AbstractRegister](https://docs.yaoquantum.org/dev/man/registers.html).
+refer to [Yao:Array Registers](https://docs.yaoquantum.org/dev/man/registers.html#Array-Registers) and [Yao:AbstractRegister](https://docs.yaoquantum.org/dev/man/registers.html).
```
Comment (Roger-luo) [inline docs/src/registers.md:67]:
```suggestion
refer to [Yao:Registers](https://docs.yaoquantum.org/dev/man/registers.html).
```

**Review by Roger-luo — COMMENTED** [review]
can we double check the links locally using 

```sh
.ci/run doc serve
```

?

**Review by Roger-luo — APPROVED** [review]
LGTM

**Pull request merged**